src/Symfony/Component/Message/Asynchronous/Routing/SenderLocator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Asynchronous\Routing;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class SenderLocator implements SenderLocatorInterface
+{
+    private $senderServiceLocator;
+    private $messageToSenderIdsMapping;
+
+    public function __construct(ContainerInterface $senderServiceLocator, array $messageToSenderIdsMapping)
+    {
+        $this->senderServiceLocator = $senderServiceLocator;
+        $this->messageToSenderIdsMapping = $messageToSenderIdsMapping;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSendersForMessage($message): array
+    {
+        $senderIds = $this->messageToSenderIdsMapping[get_class($message)] ?? $this->messageToSenderIdsMapping['*'] ?? array();
```

Comment:
perhaps move the magic `"*"` to a dedicated `SenderLocator` and call this `MessageSenderLocator`. Or wire 2 dedicated properties...

no real blocker.. but looks a bit like cheating here.

Reply:
Tbh, I think that's fine like this.